keras/src/layers/layer.py

```diff
@@ -1592,6 +1594,30 @@ def _open_name_scope(self):
             self._parent_path = current_path()
         return backend.name_scope(self.name, caller=self)
 
+    def rematerialized_activation_call_wrapper(self, layer_call):
+        """Wrapper to enable remat specifically for layers with activations.
+
+        This wrapper temporarily replaces the layer's activation function with a
+        rematerialized version before calling the original layer's `call` method
+        and restores it afterward.
+
+        Args:
+            layer_call: The original `call` method of a layer.
+            *args: Positional arguments passed to the layer's `call` method.
+            **kwargs: Keyword arguments passed to the layer's `call` method.
+
+        Returns:
+            The result of the layer's `call` method.
+        """
+
+        original_activation = self.activation
+        rematted_activation = remat.remat(original_activation)
+        self.activation = rematted_activation
+        try:
+            return layer_call
```

Comment:
This still won't work because you're returning the original call and `finally` will be called before `layer_call` is called.

Reply:
I forgot why we were restoring the activation function.